API/Controllers/CategoryController.cs

```diff
@@ -67,13 +73,33 @@ public async Task<ActionResult<IEnumerable<CategoryDto>>> GetUserCategories()
             return Ok(categories);
         }
 
+        [HttpGet("{categoryId}")]
+        public async Task<ActionResult<CategoryDto>> GetUserCategory(int categoryId)
+        {
+            var category =  await _categoryRepository.GetCategoryAsync(categoryId);
+            return Ok(category);
+        }
+
+        [HttpGet("pagination")]
+        public async Task<ActionResult<IEnumerable<CategoryDto>>> GetUserPaginatedCategories([FromQuery]UserParams userParams)
+        {
+            //var user = await _userRepository.GetUserByUsernameAsync(User.GetUsername());
+            userParams.CurrentUserName = User.GetUsername();
```

Comment:
jak to dziala. CurrentUserName jest w dto i uzupelniasz to w controllerze a userId przesylassz oddzielnie, czemu tak?

Reply:
A tak w ogole to po co ci userName w paramasach